lib/db/schemaupdater.go

```diff
@@ -590,7 +591,7 @@ func (db *schemaUpdater) updateSchemaTo11(_ int) error {
 		var putErr error
 		err := t.withHave(folder, protocol.LocalDeviceID[:], nil, true, func(fi FileIntf) bool {
 			f := fi.(FileInfoTruncated)
-			if f.IsDirectory() || f.IsDeleted() || f.IsInvalid() || f.BlocksHash == nil {
```

Comment:
This should have already been caught by the BlocksHash==nil? Maybe a len(BlocksHash)==0 instead which is less fragile?

Reply:
Yes this is indeed not an actual problem, just changed it for completeness.

Reply:
Also a len(Blocks)==0 for good measure

Reply:
Missed this: It's a `FileInfoTruncated` - no blocks there.